src/Storage/Notmuch.hs

```diff
@@ -189,6 +189,12 @@ getDatabasePath = do
   where
       decode = T.unpack . sanitiseText . decodeLenient . LB.toStrict
 
+countThreads ::
+     (MonadError Error m, MonadIO m) => T.Text -> FilePath -> m Int
```

Comment:
I'm wondering from a "library" perspective if I shouldn't pass in the configuration, since most other functions in this module expect it. Is it better for consistency sake?